05/app.js

```diff
@@ -0,0 +1,45 @@
+function Student(firstName, lastName){
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.grades = {};
+}
+
+Student.prototype.addGrade = function (subject, grade){
+     if(typeof this.grades[subject] === 'undefined') {
+        this.grades[subject] = [];
+      }
+      this.grades[subject].push(grade);
+}
+
+Student.prototype.calcGradeAvg= function(subject){
+    if(subject !== undefined){
+        const grades = this.grades[subject];
+        let sum = 0;
+        for(i=0; i< grades.length; i++){
+            sum+= grades[i];
+        }
+        return sum/grades.length
+    }
+    let sum = 0;
+    let counter = 0;
+    const subjects = Object.keys(this.grades);
+    for (let i = 0; i < subjects.length; i++) {
+        for (let j = 0; j < this.grades[subjects[i]].length; j++) {
+            sum += this.grades[subjects[i]][j];
+            console.log(`i: ${i} ; j: ${j}`)
+            console.log(typeof this.grades[subjects[i]][j])
+            counter++;
```

Comment:
oraz w tym miejscu...

Wykonujesz to samo tj. obliczasz średnią. Aby kod był jeszcze "ładniejszy" warto by było napisać osobnę metodę do obliczania średniej, a przez parametry przekazywać tablicę, dla której ma być ona obliczona.